### DISCUSSION TITLE
Recommended connection pool size and Kubernetes

### DISCUSSION THREAD

**Discussion by kbariotis** [prisma/prisma#17026] [Q&A]
Hi everyone,

thank you for writing this guide on how to [find the right size for the connection pool](https://www.prisma.io/docs/guides/performance-and-optimization/connection-management#recommended-connection-pool-size). 

What I would like to ask is, in the experience of this community, can this formula be applied on a Kubernetes environment (or any VM solution) as well or should it be adjusted somehow? What's your experience on that? 

I am considering an external connection pool but I would like to see if I can improve this setup at the moment.

As a suggestion, I think it would worth nothing that on the guide. :)

Thank you again

**Comment by pantharshit00** [comment]
Hello @kbariotis,

I will recommend reading this excellent piece by a popular Java connection pooling library: https://github.com/brettwooldridge/HikariCP/wiki/About-Pool-Sizing. It also has a video linked which is also a very great resource. 

In short, it will depend of size of your kubernetes VMs. If you are handling adequate load, you will need a pool size which can saturate the available threads that you have and for your case, threads are spread over multiple machines. This is assuming you are distributing load evenly across your kubernetes cluster.

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that haven’t had any recent activity and appear to be outdated.

If this discussion is still important to you or unresolved, we’d love to hear from you! Feel free to reopen it or start a new one with updated details.

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and being part of the community!
